tests/langchain/test_chat_utils.py

```diff
@@ -0,0 +1,123 @@
+    assert converted_response["id"] == getattr(response, "id", None)
+    assert converted_response["choices"][0]["message"]["content"] == response.content
+
+
+def test_transform_response_iter_to_chat_format_no_conversion():
+    response = [{"dict_response": "response"}]
+    converted_response = list(try_transform_response_iter_to_chat_format(response))
+    assert len(converted_response) == 1
+    assert converted_response[0] == response[0]
+
+
+def test_transform_response_iter_to_chat_format_ai_message():
+    response = ["string response"]
+    converted_response = list(try_transform_response_iter_to_chat_format(response))
+    assert len(converted_response) == 1
+    assert converted_response[0]["id"] is None
+    assert converted_response[0]["choices"][0]["delta"]["content"] == response[0]
+
+    response = [
+        AIMessage(
+            content="ai_message_response", id="123", response_metadata={"finish_reason": "done"}
+        )
+    ]
+    converted_response = list(try_transform_response_iter_to_chat_format(response))
+    assert len(converted_response) == 1
+    assert converted_response[0]["id"] == getattr(response[0], "id", None)
+    assert converted_response[0]["choices"][0]["delta"]["content"] == response[0].content
+    assert converted_response[0]["choices"][0]["finish_reason"] == "stop"
+
+    response = [
+        AIMessageChunk(
+            content="ai_message_chunk_response",
+            id="123",
+            response_metadata={"finish_reason": "done"},
+        ),
+        AIMessageChunk(
+            content="ai_message_chunk_response",
+            id="456",
+            response_metadata={"finish_reason": "stop"},
+        ),
+    ]
+    converted_response = list(try_transform_response_iter_to_chat_format(response))
+    assert len(converted_response) == 2
+    for i in range(2):
+        assert converted_response[i]["id"] == getattr(response[i], "id", None)
+        assert converted_response[i]["choices"][0]["delta"]["content"] == response[i].content
+        assert (
+            converted_response[i]["choices"][0]["finish_reason"]
+            == response[i].response_metadata["finish_reason"]
+        )
+
+
+def testtransform_request_json_for_chat_if_necessary_no_conversion():
+    model = MagicMock(spec=AgentExecutor)
+    request_json = {"messages": [{"role": "user", "content": "some_input"}]}
+    assert transform_request_json_for_chat_if_necessary(request_json, model) == (
+        request_json,
+        False,
+    )
+
+
+def testtransform_request_json_for_chat_if_necessary_conversion():
+    model = MagicMock(spec=SimpleChatModel)
+    request_json = {"messages": [{"role": "user", "content": "some_input"}]}
+
+    with patch("mlflow.langchain.utils.chat._get_lc_model_input_fields", return_value={"messages"}):
+        transformed_request = transform_request_json_for_chat_if_necessary(request_json, model)
+        assert transformed_request == (request_json, False)
```

Comment:
```suggestion
    with patch("mlflow.langchain.utils.chat._get_lc_model_input_fields", return_value={"messages"}):
        transformed_request = transform_request_json_for_chat_if_necessary(request_json, model)
        assert transformed_request == (request_json, False)
```
nit: shall we move this to the no_conversion test?